bson/src/test/unit/org/bson/json/JsonReaderTest.java

```diff
@@ -110,6 +113,16 @@ public void testDateTimeMaxBson() {
     }
 
     @Test
+    public void testDateTimeNow() {
+        long currentTime = new Date().getTime();
+        String json = "Date()";
+        bsonReader = new JsonReader(json);
+        assertEquals(BsonType.DATE_TIME, bsonReader.readBsonType());
```

Comment:
In the shell (and in JavaScript) calling Date() results in a string, not a BsonDateTime.